src/server/gc_worker/applied_lock_collector.rs

```diff
@@ -161,19 +163,20 @@ impl LockObserver {
 impl Coprocessor for LockObserver {}
 
 impl QueryObserver for LockObserver {
-    fn pre_apply_query(&self, _: &mut ObserverContext<'_>, requests: &[RaftRequest]) {
-        if !self.state.is_clean() {
+    fn post_apply_query(&self, _: &mut ObserverContext<'_>, cmd: &mut Cmd) {
+        fail_point!("notify_lock_observer_query");
+        let max_ts = self.state.load_max_ts();
+        if max_ts.is_zero() {
             return;
         }
 
-        let max_ts = self.state.load_max_ts();
-        if max_ts.is_zero() {
+        if !self.state.is_clean() {
```

Comment:
Does it make any difference by flipping the order of max_ts check and state clean check?

Reply:
Normally, `state` is always clean. Changing the order can reduce one atomic load when green GC is not running.